fendermint/vm/interpreter/src/chain.rs

```diff
@@ -228,6 +230,14 @@ where
                     }
                 }
                 ChainMessage::Signed(signed) => {
+                    // We do not need to check against the minimium base fee because the gas market
+                    // is guaranteed to be capped at it anyway.
+                    if signed.message.gas_fee_cap < gas_market.base_fee {
```

Comment:
What about doing that in tx selection for block inclusion (as a leader)? When checking for mempool inclusion only the min fee is verified.

Reply:
@cryptoAtwill do we not already exclude transactions whose gas fee cap is below the network's base fee at block production time? I was strangely not able to find that logic here: https://github.com/consensus-shipyard/ipc/blob/702d49b619623915772fe935a86e59a89744c3b1/fendermint/vm/interpreter/src/selector.rs

(Do we need a second selector?)

Reply:
@raulk I dont remember discussing that in `prepare`. I think we were mostly discussing the cometbft side of things.

Reply:
What do you mean with "the CometBFT side of things"? We agreed that we'd assign negative priorities to non-includable transactions in CheckTX, to instruct CometBFT to prefer sending us includable transactions, but we'd still have to filter out non-includable transactions in `prepare`.

Reply:
I'll fix it in this PR.

Reply:
if we need to enforce below base fee exclusion, we probably should add the other selector to enforce in fendermint just to be sure. Cometbft should have done one round of selection based on priority.